### PR TITLE
ports/esp32/boards: Add support for ESP32-S3 boards with Octal SPIRAM

### PR DESCRIPTION
**Pull request by Red-M**
These boards in espressif's documentation list which ESP32-S3s require octal or quad SPIRAM
[image](https://www.espressif.com/sites/default/files/documentation/esp32-s3_datasheet_en.pdf)

page 11, table 1.2

### CODE REVIEW

File: ports/esp32/boards/GENERIC_S3_SPIRAM_OCT/mpconfigboard.cmake [hunk 1]
```diff
@@ -0,0 +1,11 @@
+set(IDF_TARGET esp32s3)
+
+set(SDKCONFIG_DEFAULTS
+    boards/sdkconfig.base
+    boards/sdkconfig.usb
+    boards/sdkconfig.ble
+    boards/sdkconfig.240mhz
+    boards/sdkconfig.spiram_sx
```
Comment (Wind-stormger) [inline ports/esp32/boards/GENERIC_S3_SPIRAM_OCT/mpconfigboard.cmake:8]: Maybe you should copy the necessary configuration code from sdkconfig.spiram_sx to sdkconfig.spiram_oct , so you can remove the sdkconfig.spiram_sx entry in this file and avoid confusion.

**Comment by Red-M** [comment]
I'll put some more detail around why this needs to be supported as to what boards use this as an example soon, I'm a little rushed at the current time

**Comment by Wind-stormger** [comment]
ESP32-S3R8 uses  Octal SPIRAM.

**Comment by Red-M** [comment]
These boards in espressif's documentation list which ESP32-S3s require octal or quad SPIRAM
[image](https://www.espressif.com/sites/default/files/documentation/esp32-s3_datasheet_en.pdf)

page 11, table 1.2

**Comment by Red-M** [comment]
@dpgeorge What needs to be done here to get this merged?

**Comment by Red-M** [comment]
The commit message regex check doesn't work, I've included my own 2 commits and other commits from `master`
image

**Comment by Red-M** [comment]
Can this be reviewed again?

**Comment by Red-M** [comment]
Is there anything blocking this from being merged?

**Comment by Red-M** [comment]
I'm still here, waiting.

**Comment by victorallume** [comment]
I've tried compiling and flashing this PR on an ESP32-S3-WROOM1-N8R8 (8Mb OctoSPI RAM, 8Mb QuadSPI flash) (on an ESP32-S3-DevKitC-1 board), but doesn't get me to a micropython prompt, I just get the following on a repeat every 10s when the WDT times out:
```
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x8 (TG1WDT_SYS_RST),boot:0xb (SPI_FAST_FLASH_BOOT)
Saved PC:0x40375769
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3808,len:0xf24
load:0x403c9700,len:0xc44
load:0x403cc700,len:0x2d74
entry 0x403c98f4
W (29) boot.esp32s3: PRO CPU has been reset by WDT.
W (29) boot.esp32s3: APP CPU has been reset by WDT.

```
I'm compiling using `make BOARD=GENERIC_S3_SPIRAM_OCT` and using ESP-IDF v4.4.2. If I compile for GENERIC_S3 on the same fork and IDF, I get the same boot info (except the WDT lines starting with W, and a different reset cause and saved PC), but then it proceeds to a micropython prompt.

Any suggestions?

**Comment by eggfly** [comment]
Thank you for this change! I want to use this feature on LilyGo T-Embed. It uses S3 and OPI PSRAM.
https://github.com/Xinyuan-LilyGO/T-Embed

**Comment by mattytrentini** [comment]
The Espressif [ESP32-S3 Ordering Guide](https://docs.espressif.com/projects/esp-idf/en/latest/esp32s3/hw-reference/esp32s3/user-guide-devkitc-1.html#ordering-information) has a handy list of their dev boards - and which use octal SPIRAM.

Also, I have a LilyGo [T-Embed](https://www.lilygo.cc/products/t-embed), [T-RGB](https://www.lilygo.cc/products/t-rgb) and [T-Display-S3](https://www.lilygo.cc/products/t-display-s3), all of which use an ESP32-S3R8 (8MB octal PSRAM). I'll try and test at least one of them...

**Comment by eggfly** [comment]
[lilygo-t-embed-mpy-with7789.bin.zip](https://github.com/micropython/micropython/files/10120671/lilygo-t-embed-mpy-with7789.bin.zip)
This file is thre firmware for Lilygo T-Embed with ST7789 and OPI PSRAM.

**Comment by dpgeorge** [comment]
Thanks for the contribution and sorry this got left for so long.

Octal SPIRAM seems to be in a few boards now so will be good to have support for it, and I think a generic board like the one in this PR is a good way to do it.

One thing to change: would be good to enable BLE.  That was left out by accident on existing S3 boards, but enabled in 8a91c719668b9aa652d1c055208a9b04e5c682df and b705732d5f385721f5c50d9735252bfe10f180ec.

@Red-M are you able to enable BLE on this PR?  Then it can be merged.

**Comment by Red-M** [comment]
I can take a look tomorrow to enable BLE.

**Comment by Red-M** [comment]
I already had the `boards/sdkconfig.ble` line for the cmake file, this should be good to go now.

**Comment by dpgeorge** [comment]
Thanks for updating.  Squashed and merged in 1290329415b2a61af2f5b42e1e9bc922d7915d48

**Comment by mattytrentini** [comment]
A little late...but I just successfully tested `GENERIC_S3_SPIRAM_OCT` on a [Wemos S3 PRO](https://www.wemos.cc/en/latest/s3/s3_pro.html):

```bash
>mpremote connect com13
Connected to MicroPython at com13
Use Ctrl-] to exit this shell

>>>
MPY: soft reboot
MicroPython v1.19.1-777-g129032941-dirty on 2022-12-19; ESP32S3 module (spiram octal) with ESP32S3
Type "help()" for more information.
>>> import micropython
>>> micropython.mem_info()
stack: 704 out of 15360
GC: total: 8196480, used: 1584, free: 8194896
 No. of 1-blocks: 27, 2-blocks: 8, max blk sz: 18, max free sz: 512168
```

Thanks @Wind-stormger!

**Pull request closed without merging**